### PR TITLE
Add solution for Add Two Integers problem

### LINKED ISSUE #7: Add Two Integers
## Issue Overview

Implement a solution for the "Add Two Integers" LeetCode problem to practice basic Rust syntax and function implementation.

## Feature Details

Create a Rust implementation for [Add Two Integers](https://leetcode.com/problems/add-two-integers/).

Problem summary:
- Given two integers num1 and num2, return the sum of the two integers.
- Constraints: -100 <= num1, num2 <= 100

The solution should:
- Implement the function using Rust's basic arithmetic operations
- Ignore LeetCode's provided function signature and create a more idiomatic Rust version if desired
- Include appropriate tests

## Expected Outcome

A working Rust implementation that correctly adds two integers, with clean, idiomatic code and proper test coverage.

### PR DESCRIPTION
**Pull request by zmrdltl**
## Related Issues

closes #7 

## PR Description

This PR adds a solution for the LeetCode "Add Two Integers" problem.

The implementation includes:
- A well-documented function with proper doctest examples
- Error handling for integer overflow using Result
- Comprehensive test suite covering normal cases and edge cases

The project is structured as a library crate, with:
1. Implementation in src/numbers/add_two_integers.rs
2. Tests within the same file using Rust's standard test module pattern
3. Documentation following Rust's conventions, including Examples and Errors sections

This establishes a pattern that can be followed for future LeetCode solutions.

### CODE REVIEW

**Review by kimhanbeom — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "rust-leetcode"
+version = "0.1.0"
+edition = "2021"
```
Comment (sophie-cluml) [inline Cargo.toml:4]: 2024를 사용하도록 해주세요.
